src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectRpm.hpp

```diff
@@ -186,22 +248,27 @@ class VersionObjectRpm final : public IVersionObject
     static bool match(const std::string& version, Rpm& output)
     {
         std::smatch parserMatches;
-        if ((std::regex_match(version, parserMatches, m_parserRegex) == false) || (parserMatches.size() != 4))
+        if ((std::regex_search(version, parserMatches, m_parserRegex)) && (parserMatches.size() == 4))
+        {
+            output.epoch = parserMatches[1].str().empty() ? 0 : std::stoi(parserMatches[1].str());
+            output.version = parserMatches[2].str();
+            output.release = parserMatches[3].str();
+            return true;
+        }
+        else
         {
             return false;
         }
```

Comment:
Nitpick: Unnecesary else
```suggestion

        return false;
```

Reply:
Deleted, old version.